### PR TITLE
Add replication concepts and reference guide

### PR DESCRIPTION
**Pull request by the-joksim**
* added replication sections to `Concepts` and `Reference guide`

### CODE REVIEW

File: products/memgraph/upcoming/concepts/replication.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+## High Level Context
+
+Replication is a method that ensures that multiple database instances are
+storing the same data. To enable replication, there must be at least two
+instances of Memgraph in a cluster. Each instance has one of either two roles:
+main or replica. The main instance is the instance that accepts writes to the
+database and replicates its state to the replicas. In a cluster, there can only
+be one main. There can be one or more replicas. None of the replicas will accept
+write queries, but they will always accept read queries (there is an exception
+to this rule and is described below).  ach instance will always be reachable
```
Comment (gitbuda) [inline products/memgraph/upcoming/concepts/replication.md:12]: Each?

File: products/memgraph/upcoming/concepts/replication.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+### Addition of a New Replica
+
+Each replica, when added to the cluster (in any mode), will first start out as
+an asynchronous replica. That will allow replicas that have fallen behind to
+first catch-up to the current state of the database. When the replica is in a
+state that it isn't lagging behind the main it will then be promoted (in a brief
+stop-the-world operation) to a semi-synchronous or synchronous replica. Slaves
+that are added as asynchronous replicas will remain asynchronous.
+
+
```
Comment (gitbuda) [inline products/memgraph/upcoming/concepts/replication.md:77]: Extra new lines.

File: products/memgraph/upcoming/concepts/replication.md (hunk 1 above)
Comment (the-joksim) [reply products/memgraph/upcoming/concepts/replication.md:12]: Sure.

File: products/memgraph/upcoming/concepts/replication.md (hunk 2 above)
Comment (the-joksim) [reply products/memgraph/upcoming/concepts/replication.md:77]: Whoops.

File: products/memgraph/upcoming/concepts/replication.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+that the data within the cluster is consistent, but you don't want the main
+to grind to a halt when you have a non-responsive replica.
+
+### Addition of a New Replica
+
+Each replica, when added to the cluster (in any mode), will first start out as
+an asynchronous replica. That will allow replicas that have fallen behind to
+first catch-up to the current state of the database. When the replica is in a
+state that it isn't lagging behind the main it will then be promoted (in a brief
+stop-the-world operation) to a semi-synchronous or synchronous replica. Slaves
```
Comment (antonio2368) [inline products/memgraph/upcoming/concepts/replication.md:74]: remove the `in a brief stop-the-world operation`
Also, change `Slaves` :smile:

**Review by antonio2368 — APPROVED** [review]

**Review by g-despot — APPROVED** [review]

**Pull request merged**